### ISSUE TITLE
torch.nn.functional.layer_norm returns an incorrect second-order JVP

### ISSUE THREAD

**Issue by ALinrunrun** [pytorch/pytorch#196700]
### 🐛 Describe the bug

`torch.nn.functional.layer_norm` returns `-0.38487405661968344` for a nested forward-mode second derivative whose mathematically correct value is `0.7749142079590942` in `float64`.
The normalized shape and affine parameters are fixed, and `eps = 0.25`. The scalar output simplifies to `-t / sqrt(1 + t**2)`, which is smooth. Differentiating twice at `t = 0.7` gives `0.7749142079590942`. PyTorch’s nested JVP has the opposite sign. The forward value matches.
This issue reproduces on CPU, so CUDA availability does not affect the result.
```python
import torch
dtype = torch.float64
def f(t):
    zero = torch.zeros((), dtype=t.dtype)
    x = torch.stack((t, zero)).reshape(1, 2)
    weight = torch.tensor([1, 2], dtype=t.dtype)
    bias = torch.zeros((2,), dtype=t.dtype)
    return torch.nn.functional.layer_norm(
        x, normalized_shape=[2], weight=weight, bias=bias, eps=0.25
    ).sum()
def jvp1(t):
    return torch.func.jvp(f, (t,), (torch.ones_like(t),))[1]
t = torch.tensor(0.7, dtype=dtype)
_, actual = torch.func.jvp(jvp1, (t,), (torch.ones_like(t),))
expected = torch.tensor(0.7749142079590942, dtype=dtype)
print("actual:")
print(actual)
print("expected:")
print(expected)
```
### Actual vs expected result
```text
actual:
-0.38487405661968344
expected:
0.7749142079590942
```
The discrepancy occurs in the nested (second-order) forward-mode derivative. The forward value is `-0.5734623443633283` and matches `-t / sqrt(1 + t**2)`.

### Versions

```text
PyTorch version: 2.12.0+cu130
Is debug build: False
CUDA used to build PyTorch: 13.0
ROCM used to build PyTorch: N/A
OS: Ubuntu 24.04.3 LTS (x86_64)
Python version: 3.13.13 | packaged by Anaconda, Inc.
Python platform: Linux x86_64
Is CUDA available: False
GPU models:
  GPU 0: NVIDIA RTX 6000 Ada Generation
  GPU 1: NVIDIA RTX 6000 Ada Generation
Nvidia driver version: 570.211.01
CPU:
  Model name: AMD Ryzen Threadripper PRO 7985WX 64-Cores
  CPU(s): 128
Versions of relevant libraries:
[pip3] numpy==2.4.6
[pip3] torch==2.12.0
[pip3] triton==3.7.0
```

cc @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki

**Comment by MDSALMANSHAMS** [comment]
Flagging that this isn't isolated — the same reporter filed 7 more nested-JVP / second-order forward-AD correctness bugs the same day, all still bot-triaged only: #196697 (`linalg.slogdet`), #196698 (`linalg.householder_product`), #196699 (`batch_norm`), #196701 (`cumprod`), #196702 (`scatter_reduce`), #196703 (`sinc`), #196707 (`triangular_solve`). Most share `module: correctness (silent)` + `module: forward ad`. Each repro derives the expected value analytically (not just "looks wrong"), so this reads as a systematic gap in nested-JVP/double-backward formulas across several ops rather than 8 unrelated one-offs.

Could this set get `actionable`, or get routed to whoever owns nested forward-AD correctness? Given the shared root class, it may be worth tracking as a group rather than triaging each independently.
